miden-lib/src/transaction/errors.rs

```diff
@@ -123,7 +123,7 @@ impl std::error::Error for TransactionKernelError {}
 // TRANSACTION EVENT PARSING ERROR
 // ================================================================================================
 
-#[derive(Debug, Clone, Eq, PartialEq)]
+#[derive(Debug)]
 pub enum TransactionEventParsingError {
```

Comment:
Should we change impl `Error` to come from `core` (i.e., on line 146 below) for this and other similar errors? I know full migration to `thiserror` would be in follow-up PRs, but maybe this is simple enough to do now.